addons/account/models/account_bank_statement.py

```diff
@@ -884,10 +886,16 @@ def process_reconciliation(self, counterpart_aml_dicts=None, payment_aml_rec=Non
                 raise UserError(_('A selected move line was already reconciled.'))
             if isinstance(aml_dict['move_line'], pycompat.integer_types):
                 aml_dict['move_line'] = aml_obj.browse(aml_dict['move_line'])
+
+        account_types = self.env['account.account.type']
         for aml_dict in (counterpart_aml_dicts + new_aml_dicts):
             if aml_dict.get('tax_ids') and isinstance(aml_dict['tax_ids'][0], pycompat.integer_types):
                 # Transform the value in the format required for One2many and Many2many fields
                 aml_dict['tax_ids'] = [(4, id, None) for id in aml_dict['tax_ids']]
+
+            user_type_id = self.env['account.account'].browse(aml_dict.get('account_id')).user_type_id
```

Comment:
in python code (not in fields definition) use `user_type` intead `user_type_id` (by convention `user_type` is a recordset, `user_type_id` is an integer)